File: LeetCode_rename/557_ReverseWordsinaStringIII.py
# https://leetcode.com/problems/reverse-words-in-a-string-iii/submissions/
# 2021 11.25

class Solution:
    def reverseWords(self, s: str) -> str:
        # # v1 split and join
        # list1=s.split(' ')
        # for i in range(len(list1)):
        #     list1[i]=list1[i][::-1]
        # return ' '.join(list1)
        
        # v1.1 copied oneliner split and join
        return ' '.join(x[::-1] for x in s.split())
        
        # # v1.2 copied oneliner using lambda
        # return " ".join(map(lambda x: x[::-1], s.split()))

File: LeetCode/557_ReverseWordsinaStringIII.go
// https://leetcode.com/problems/reverse-words-in-a-string-iii/submissions/
// 2022年07月05日 16:39:29

import (
    "fmt"
    "strings"
)
func reverseWords(s string) string {
    list_s := strings.Split(s, " ")
    for i:=0;i<len(list_s);i++{
        list_s[i]=reverse(list_s[i])
    }
    return strings.Join(list_s," ")
}

func reverse(s string) string{
    ls := strings.Split(s, "")
    for i,j:=0,len(s)-1 ; i<j ; i,j=i+1,j-1{
        ls[i],ls[j]=ls[j],ls[i]
    }
    return strings.Join(ls,"")
}